pkg/dataflatten/json.go

```diff
@@ -1,25 +1,49 @@
 package dataflatten
 
 import (
+	"bufio"
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"os"
 )
 
 func JsonFile(file string, opts ...FlattenOpts) ([]Row, error) {
-	rawdata, err := os.ReadFile(file)
+	f, err := os.Open(file)
+	defer f.Close()
 	if err != nil {
 		return nil, err
 	}
-	return Json(rawdata, opts...)
+
+	return Json(bufio.NewReader(f), opts...)
 }
 
-func Json(rawdata []byte, opts ...FlattenOpts) ([]Row, error) {
-	var data interface{}
+func Json(r io.Reader, opts ...FlattenOpts) ([]Row, error) {
+	decoder := json.NewDecoder(r)
+	var objects []interface{}
+
+	for {
+		var object interface{}
+		err := decoder.Decode(&object)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+
+			return nil, fmt.Errorf("unmarshalling json: %w", err)
+		}
+		objects = append(objects, object)
```

Comment:
This presents an interesting question. As written here, this will parse well formed jsonl data. But if we replaced the return with a continue, we'd also be able to parse:
```
Some test that might, happen 
{ a: 1, b: 2 }
To have a json blob in the middle
```

But I'm not sure that's a good idea.

Reply:
I tried messing with that ... it was very easy to hit infinite loop ... it might be possible, but not sure it's worth the effort